pymatgen/entries/__init__.py

```diff
@@ -114,5 +141,32 @@ def as_dict(self) -> dict:
             "@module": self.__class__.__module__,
             "@class": self.__class__.__name__,
             "energy": self._energy,
-            "composition": self.composition.as_dict(),
+            "composition": self.composition.as_dict()
         }
+
+    def __eq__(self, other):
+        # NOTE Scaled duplicates i.e. physically equivalent materials
+        # are not equal unless normalized separately
+        if self is other:
+            return True
+
+        if isinstance(other, self.__class__):
+            if not np.allclose(self.energy, other.energy):
+                return False
+            if self.composition != other.composition:
+                return False
+
+            # NOTE this is not performant and should be preceeded
```

Comment:
Responding to @shyamd's concern about equality checking and floating point precision, would something like the following work?

```
self_dict = self.as_dict()
other_dict = other.as_dict()

def is_eq(v1, v2):
    """Will use np.allclose for numerical values,
    otherwise will use default __eq__"""
    if isinstance(v1, Number) and isinstance(v2, Number):
        return np.allclose(v1, v2)
    else:
        return v1 == v2

return all([is_eq(other_dict[k], v) for k, v in self_dict.items()])
```

Reply:
Right, that's a good point -- my hope here was that the check the line above that both `self` and `other` are the same class would mean that both dicts should be guaranteed to have the same keys (technically, the check above uses `isinstance` so would also pass for subclasses, so perhaps this could be tightened).

Reply:
Yep I call `isinstance` first so they would have the same keys but they could "physically" correspond to the same material but one `ComputedEntry` might have the optional `parameters` and the second might have `parameters` be `None`. Currently this means they would not be equal due to a metadata attribute. I think that's probably how it should be as it's closer to how other things are but I just wanted to try highlight it explicitly.